File: modules/EnchantmentsEnhance-Plugin/src/main/java/org/pixeltime/enchantmentsenhance/listener/VanillaEnchantListener.java
package org.pixeltime.enchantmentsenhance.listener;


import com.lgou2w.ldk.bukkit.compatibility.XMaterial;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.block.Action;
import org.bukkit.event.player.PlayerInteractEvent;
import org.pixeltime.enchantmentsenhance.gui.GUIAbstract;
import org.pixeltime.enchantmentsenhance.gui.GUIManager;
import org.pixeltime.enchantmentsenhance.gui.menu.MainMenu;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;


public class VanillaEnchantListener implements Listener {
    @EventHandler(priority = EventPriority.HIGHEST)
    public void onOpenEnchantmentTable(PlayerInteractEvent event) {
        if (event.getClickedBlock() != null && event.getClickedBlock().getType() == XMaterial.ENCHANTING_TABLE.toBukkit()) {
            if (SettingsManager.config.getString("openMethod").equalsIgnoreCase("LEFT_CLICK")) {
                if (event.getAction() == Action.LEFT_CLICK_BLOCK) {
                    event.setCancelled(true);
                    new MainMenu(event.getPlayer()).open();
                }
            } else if (SettingsManager.config.getString("openMethod").equalsIgnoreCase("RIGHT_CLICK")) {
                if (event.getAction() == Action.RIGHT_CLICK_BLOCK) {
                    event.setCancelled(true);
                    GUIAbstract gui = GUIManager.getMap().get(event.getPlayer().getName());
                    if (gui == null)
                        new MainMenu(event.getPlayer()).open();
                }
            } else if (SettingsManager.config.getString("openMethod").equalsIgnoreCase("SHIFT_AND_LEFT_CLICK")) {
                if (event.getAction() == Action.LEFT_CLICK_BLOCK && event.getPlayer().isSneaking()) {
                    event.setCancelled(true);
                    new MainMenu(event.getPlayer()).open();
                }
            } else if (SettingsManager.config.getString("openMethod").equalsIgnoreCase("SHIFT_AND_RIGHT_CLICK")) {
                if (event.getAction() == Action.RIGHT_CLICK_BLOCK && event.getPlayer().isSneaking()) {
                    event.setCancelled(true);
                    GUIAbstract gui = GUIManager.getMap().get(event.getPlayer().getName());
                    if (gui == null)
                        new MainMenu(event.getPlayer()).open();
                }
            }
        }
    }
}

File: modules/EnchantmentsEnhance-Plugin/src/main/kotlin/org/pixeltime/enchantmentsenhance/listener/VanillaEnchantListener.kt
package org.pixeltime.enchantmentsenhance.listener


import com.lgou2w.ldk.bukkit.compatibility.XMaterial
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.block.Action
import org.bukkit.event.player.PlayerInteractEvent
import org.pixeltime.enchantmentsenhance.gui.menu.MainMenu
import org.pixeltime.enchantmentsenhance.manager.SettingsManager


class VanillaEnchantListener : Listener {
    @EventHandler(priority = EventPriority.HIGHEST)
    fun onOpenEnchantmentTable(event: PlayerInteractEvent) {
        if (event.clickedBlock != null && event.clickedBlock!!.type == XMaterial.ENCHANTING_TABLE.toBukkit()) {
            if (SettingsManager.config.getString("openMethod")!!.equals("LEFT_CLICK", ignoreCase = true)) {
                if (event.action == Action.LEFT_CLICK_BLOCK) {
                    event.isCancelled = true
                    MainMenu(event.player).open()
                }
            } else if (SettingsManager.config.getString("openMethod")!!.equals("RIGHT_CLICK", ignoreCase = true)) {
                if (event.action == Action.RIGHT_CLICK_BLOCK) {
                    event.isCancelled = true
                    MainMenu(event.player).open()
                }
            } else if (SettingsManager.config.getString("openMethod")!!.equals("SHIFT_AND_LEFT_CLICK", ignoreCase = true)) {
                if (event.action == Action.LEFT_CLICK_BLOCK && event.player.isSneaking) {
                    event.isCancelled = true
                    MainMenu(event.player).open()
                }
            } else if (SettingsManager.config.getString("openMethod")!!.equals("SHIFT_AND_RIGHT_CLICK", ignoreCase = true)) {
                if (event.action == Action.RIGHT_CLICK_BLOCK && event.player.isSneaking) {
                    event.isCancelled = true
                    MainMenu(event.player).open()
                }
            }
        }
    }
}